pkg/internal/storage/repo/outputs.go

```diff
@@ -135,3 +135,62 @@ func (o *Outputs) ListAndCountOutputs(ctx context.Context, filter entity.ListOut
 
 	return slices.Map(outputs, o.mapModelToTableOutput), total, nil
 }
+
+func (o *Outputs) FindInputsAndOutputsWithBuckets(ctx context.Context, txIDs []uint) (inputs map[uint][]*wdk.TableOutput, outputs map[uint][]*wdk.TableOutput, err error) {
+	if len(txIDs) == 0 {
+		return nil, nil, nil
+	}
+
+	var allOutputs []*models.Output
+	if err := o.db.WithContext(ctx).
+		Preload("Basket").
+		Where("transaction_id IN ? OR spent_by IN ?", txIDs, txIDs).
+		Find(&allOutputs).Error; err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch inputs/outputs: %w", err)
+	}
+
+	inputMap := make(map[uint][]*wdk.TableOutput)
+	outputMap := make(map[uint][]*wdk.TableOutput)
+
+	for _, out := range allOutputs {
+		tableOut := o.mapModelToTableOutput(out)
+		if out.GetSpentBy() != 0 {
+			inputMap[out.GetSpentBy()] = append(inputMap[out.GetSpentBy()], tableOut)
+		}
```

Comment:
This way we could skip the `GetSpentBy` method, which sill I don't like. 
```suggestion
		if out.SpentBy != nil {
			inputMap[*out.SpentBy] = append(inputMap[*out.SpentBy], tableOut)
		}
```

Why I don't like the `GetSpentBy` metod? Because:
- `0` can ba valid ID in database (depends on DB settings of autoincrement)